### PR TITLE
added sync flag

### PR DESCRIPTION
**Pull request by adp-atea**

### CODE REVIEW

**Comment by adp-atea** [comment]
Probably will be more changes with this task
https://github.com/orgs/atviriduomenys/projects/3/views/5?pane=issue&itemId=18394733

File: notes/push/sync.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+#|                     'schema': '5',
+#|                 },
+#|                 'message': 'Given value already exists.',
+#|                 'template': 'Given value already exists.',
+#|                 'type': 'property',
+#|             },
+#|         ],
+#|     }
+# FIXME: It looks, that sync was not done, because an attempt to create an
+#        existing City was made.
```
Comment (sirex) [inline notes/push/sync.sh:87]: If `--sync` would work correctly, this error would not happen.

File: notes/push/sync.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+#| +--------------------------------------+------------------------------------------+----------+----------------------------+-------+--------------------------------------------------------------+-------------+
+#| | 85828ce4-5264-4fc0-bd9a-699ee91f6298 | 866314955098a3fa463e7ad8e224437bb6860094 |          | 2023-08-04 15:24:01.083430 | 1     | {"_type": "push/sync/City", "_id": "85828ce4-5264-4fc0-bd9a- |             |
+#| |                                      |                                          |          |                            |       | 699ee91f6298", "id": 1, "name": "Vilnius", "_op": "insert"}  |             |
+#| +--------------------------------------+------------------------------------------+----------+----------------------------+-------+--------------------------------------------------------------+-------------+
+# FIXME: Yes, an erros was registerend and nothing was pushed. Instead of using VILNIUS=646d0f3f-1d6d-493e-b1b9-b6d1472d36fb, a new id was generated.
```
Comment (sirex) [inline notes/push/sync.sh:96]: This, shows, that `--sync` was not implemented or does not work as intended.

File: notes/push/sync.sh [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+sqlite3 $BASEDIR/keymap.db "SELECT * FROM '"$DATASET"/City'"
+#| +--------------------------------------+------------------------------------------+-------+
+#| |                 key                  |                   hash                   | value |
+#| +--------------------------------------+------------------------------------------+-------+
+#| | 85828ce4-5264-4fc0-bd9a-699ee91f6298 | bf8b4530d8d246dd74ac53a13471bba17941dff7 |       |
+#| +--------------------------------------+------------------------------------------+-------+
+# FIXME: It looks, that a new id was generated, instead of using existing VILNIUS=646d0f3f-1d6d-493e-b1b9-b6d1472d36fb.
```
Comment (sirex) [inline notes/push/sync.sh:104]: This, shows, that `--sync` was not implemented or does not work as intended.

**Review by sirex — CHANGES_REQUESTED** [review]
There are no signs that `--sync` works, please see testing notes.

File: notes/push/sync.sh (hunk 1 above)
Comment (adp-atea) [reply notes/push/sync.sh:87]: Decided that this task will be split into push state sync and keymap sync, currently only keymap part is done

**Review by sirex — APPROVED** [review]

**Pull request merged**